src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const groupedMap = items.reduce((acc: GroupsMap<T>, el) => {
+    const valueOfKey = el[key] as string;
```

Comment:
Avoid using typecasting with `as` in TS
```suggestion
    const valueOfKey = String(el[key]);
```